tracer/test/benchmarks/Benchmarks.Trace/AppSecBodyBenchmark.cs

```diff
@@ -0,0 +1,247 @@
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using Datadog.Trace;
+using Datadog.Trace.AppSec;
+using Datadog.Trace.AppSec.Waf;
+using Datadog.Trace.Configuration;
+using System.Diagnostics;
+#if NETFRAMEWORK
+using System.Web;
+#else
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+#endif
+
+namespace Benchmarks.Trace
+{
+    [MemoryDiagnoser]
+    public class AppSecBodyBenchmark
+    {
+        private Security security;
+        private readonly ComplexModel complexModel = new()
+        {
+            Age = 12,
+            Gender = "Female",
+            Name = "Tata",
+            LastName = "Toto",
+            Address = new Address
+            {
+                Number = 12,
+                City = new City { Name = "Paris", Country = new Country { Name = "France", Continent = new Continent { Name = "Europe", Planet = new Planet { Name = "Earth" } } } },
+                IsHouse = false,
+                NameStreet = "lorem ipsum dolor sit amet"
+            },
+            Address2 = new Address
+            {
+                Number = 15,
+                City = new City
+                {
+                    Name = "Madrid",
+                    Country = new Country
+                    {
+                        Name = "Spain",
+                        Continent = new Continent
+                        {
+                            Name = "Europe",
+                            Planet = new Planet { Name = "Earth" }
+                        }
+                    }
+                },
+                IsHouse = true,
+                NameStreet = "lorem ipsum dolor sit amet"
+            },
+            Dogs = new List<Dog> {
+                    new Dog { Name = "toto", Dogs = new List<Dog> { new Dog { Name = "titi" }, new Dog { Name = "titi" } } },
+                    new Dog { Name = "toto", Dogs = new List<Dog> { new Dog { Name = "tata" }, new Dog { Name = "tata" } } },
+                    new Dog { Name = "tata", Dogs = new List<Dog> { new Dog { Name = "titi" }, new Dog { Name = "titi" }, new Dog { Name = "tutu" } } }
+                    }
+        };
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            security = Security.Instance;
+        }
```

Comment:
nit: rather than using `GlobalSetup` we typically use a static constructor. Nothing wrong with this AFAIK, just wondered if there was a specific reason for this approach?

Reply:
there was no reason , as I had problems initializing Security, I tried this and left it 😅

Reply:
fixed in https://github.com/DataDog/dd-trace-dotnet/pull/2537/commits/2dd558752b586980c3b56b32ef30423f75f009c1